Using "map" as a variable name generate invalid C code

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux version 5.14.16-301.fc35.x86_64 (mockbuild@bkernel01.iad2.fedoraproject.org) (gcc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1), GNU ld version 2.37-10.fc35) #1 SMP Wed Nov 3 13:55:42 UTC 2021
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 5800X 8-Core Processor
CC version: cc (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)

getwd: /home/sheat/Documents/Projects/v/experiments
vmodules: /home/sheat/.vmodules
vroot: /home/sheat/Documents/Tools/v
vexe: /home/sheat/Documents/Tools/v/v
vexe mtime: 2021-11-15 13:12:49
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 6937074.a8a1e93

Git version: git version 2.33.1
Git vroot status: weekly.2021.43-142-ga8a1e938
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg borked.v`
```v
module main

fn main()
{
	map := map[string]string
	println("nope" in map)
}
```

**What did you expect to see?**

A compilation issue, as map is a type.


**What did you see instead?**
No error(s) from V and generated invalid C code.
Expecting an error like "invalid use of reserved type `map` as a variable name"'


```
==================
/tmp/v_1000/borked.12010874479825168407.tmp.c:10662: error: identifier expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -32,7 +32,7 @@ const (
 	array_builtin_methods           = ['filter', 'clone', 'repeat', 'reverse', 'map', 'slice',
 		'sort', 'contains', 'index', 'wait', 'any', 'all', 'first', 'last', 'pop']
 	reserved_type_names             = ['bool', 'i8', 'i16', 'int', 'i64', 'byte', 'u16', 'u32',
-		'u64', 'f32', 'f64', 'string', 'rune']
+		'u64', 'f32', 'f64', 'map', 'string', 'rune']
 	vroot_is_deprecated_message     = '@VROOT is deprecated, use @VMODROOT or @VEXEROOT instead'
 )
 

--- a/vlib/v/checker/tests/invalid_variable_name_err.out
+++ b/vlib/v/checker/tests/invalid_variable_name_err.out
@@ -31,4 +31,11 @@ vlib/v/checker/tests/invalid_variable_name_err.vv:14:2: error: invalid use of re
    14 |     f64 := 2.22
       |     ~~~
    15 |     println(f64)
-   16 | }
+   16 |
+vlib/v/checker/tests/invalid_variable_name_err.vv:17:2: error: invalid use of reserved type `map` as a variable name
+   15 |     println(f64)
+   16 |
+   17 |     map := map[string]string
+      |     ~~~
+   18 |     println(map)
+   19 | }

--- a/vlib/v/checker/tests/invalid_variable_name_err.vv
+++ b/vlib/v/checker/tests/invalid_variable_name_err.vv
@@ -13,4 +13,7 @@ fn main() {
 
 	f64 := 2.22
 	println(f64)
+
+	map := map[string]string
+	println(map)
 }

